kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -82,7 +84,9 @@ public WatchConnectionManager(final OkHttpClient client, final BaseOperation<T,
     this.reconnectLimit = reconnectLimit;
     this.websocketTimeout = websocketTimeout;
 
-    this.clonedClient = client.newBuilder().readTimeout(this.websocketTimeout, TimeUnit.MILLISECONDS).build();
+    Config aConfig = baseOperation.getConfig();
+    aConfig.setRequestTimeout((int)this.websocketTimeout);
+    this.clonedClient = HttpClientUtils.createHttpClient(baseOperation.getConfig());
```

Comment:
Now it is not cloned anymore. I will use other name like `client`

Reply:
I don't like the idea of mutating the configuration object (it possilby mutates the global config instance of the client). I would prefer creating a new instance of the configuration object. Something like:

    Config requestConfig = new ConfigBuilder(baseOperation.getConfig()).withReadtTimeout(....).build();

An other option worth exploring (not sure if applicable here). is to use the withRequestConfig() thing that we have and allows you to execute a single request with a custom configuration object. As I said it might not be applicable here. So no need to put too much effort in this.

In a perfect world none of our object should have setters. Who knows maybe in 4.x....